### PR TITLE
PAT repo get api for dpu

### PR DESCRIPTION
**Pull request by tapishr**
Corresponding DPU pr - https://github.com/Alokit-Innovations/vibi-dpu/pull/138

- [x] Build Running
- [x] Manually tested

## Summary by CodeRabbit

- **New Features**
	- Enhanced data retrieval by introducing an API endpoint to fetch user repositories based on topic and provider.

### CODE REVIEW

File: pages/api/dpu/repos.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    const { topicId, provider } = req.query;
+    if (!topicId || !provider 
+        || Array.isArray(topicId) || Array.isArray(provider)
+        || topicId.length === 0 || provider.length === 0) {
+        res.status(400).json({ error: 'Invalid get request body' });
+		return;
+    }
+    const repoList: RepoIdentifier[] = await getUserRepositoriesByTopic(topicId, provider);
+	if (repoList.length == 0) {
+		return res.status(500).json({error: 'Unable to get repo list'});
```
[the remark this answers is not in the document]
Comment (MuskanPaliwal) [reply pages/api/dpu/repos.ts:15]: @tapishr I guess rabbit is correct here

File: utils/db/repos.ts [hunk 2]
```diff
@@ -38,6 +38,25 @@ export const getRepos = async (allRepos: RepoIdentifier[]) => {
...
+				repo_name: row.repo_name,
+				repo_owner: row.repo_owner,
+				repo_provider: row.repo_provider,
+			}));
+		})
+		.catch((err: Error) => {
+			console.error(`[db/getUserRepositoriesByTopic] Could not get repos for topic id ${topicId}`, err);
+			return [];
+		});
+	return repos;
```
[the remark this answers is not in the document]
Comment (MuskanPaliwal) [reply utils/db/repos.ts:57]: @tapishr I find this a better practice

**Review by MuskanPaliwal — DISMISSED** [review]
Good to go from my end. Just a few changes I would want you to look at

File: pages/api/dpu/repos.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tapishr) [reply pages/api/dpu/repos.ts:15]: Sounds good, making changes

File: utils/db/repos.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (tapishr) [reply utils/db/repos.ts]: Yes, I used the `$1` type syntax but it was not formatting the values for some reason. Didn't go too deep into it as this approach worked. Also provider can only be either github or bitbucket and topic id is generated by us, so not much of a security threat for now

**Review by avikalpg — APPROVED** [review]

**Pull request merged**